### PR TITLE
Add Client.SetConn and NewClientConn

### PR DESCRIPTION
**Pull request by cmars**
Need to be able to set the connection from other packages for mocking.

### CODE REVIEW

File: statsd.go [hunk 1]
```diff
@@ -71,6 +82,16 @@ func (c *Client) SetErrorFunc(f func(err error)) {
 	c.mu.Unlock()
 }
 
+// SetConn sets the connection to which stats will be sent.
+func (c *Client) SetConn(conn io.WriteCloser) {
```
Comment (rogpeppe) [inline statsd.go]: Perhaps SetWriter?

File: statsd.go [hunk 2]
```diff
@@ -57,6 +57,17 @@ func NewClient(hostPort string) (*Client, error) {
 	return c, nil
 }
 
+// NewClientConn creates a new statsd client that
+// will send stats to the given connection.
+func NewClientConn(conn io.WriteCloser) (*Client, error) {
```
Comment (rogpeppe) [inline statsd.go]: NewClientWithWriter ?

**Comment by rogpeppe** [comment]
LGTM but I'd suggest updating the tests to use the public API now rather than delving into the data structure.

**Comment by cmars** [comment]
Thanks! Will update tests in a followup.

**Pull request merged**